ee/server/service/software_installers.go

```diff
@@ -64,6 +64,18 @@ func (svc *Service) UploadSoftwareInstaller(ctx context.Context, payload *fleet.
 	// Update $PACKAGE_ID in uninstall script
 	preProcessUninstallScript(payload)
 
+	if err := svc.ds.ValidateEmbeddedSecrets(ctx, []string{payload.InstallScript}); err != nil {
```

Comment:
Yes it's less efficient to validate each script individually, but being since this is for interactive use, being able to tell the user which script has an invalid secret is going to be helpful